ffi/derive/src/wrapper.rs

```diff
@@ -225,8 +220,12 @@ pub fn wrap_as_opaque(mut input: syn::DeriveInput) -> TokenStream {
     let ref_inner = quote!(*const iroha_ffi::Extern);
     let ref_mut_inner = quote!(*mut iroha_ffi::Extern);
 
-    let shared_fns = gen_shared_fns(&input);
-    let attrs = wrapper_attributes(input.attrs);
+    let shared_fns = gen_shared_fns(emitter, &input);
+    // TODO: which attributes do we need to keep?
+    // in darling there is mechanism to forwards attrs, but it needs to be an whitelist
+    // it seems that as of now no such forwarding needs to take place
+    // so we just drop all attributes
+    let attrs = Vec::<Attribute>::new();
```

Comment:
Ideally it would, but it's not easy to implement such a mechanism. `darling` is tailored to derive macros, in which you get the unknown attribute handling for free from rustc: you have to specify which attributes macros you handle and rustc errors out if some attribute is unused.

For non-derive style macros, however, there's no mechanism to track this. This might be a good area of improvement, but IMO this PR is pretty large as it is